src/core/graphTestUtil.js

```diff
@@ -262,3 +268,24 @@ export function advancedGraph(): {|
   };
   return {nodes, edges, graph1, graph2};
 }
+
+export function testWeightedGraph(
+  nodesWithWeights: NodeWithWeight[],
+  edgesWithWeights: EdgeWithWeight[]
+): WeightedGraph {
```

Comment:
Note: This API makes it impossible to construct graphs that have prefix-level weights that aren't bound to any node in particular.